File: cases/taint/php/nested-access-path-negative/nested_access_path_negative.php
<?php
class Inner
{
    public string $value = "clean";
    public string $other = "clean";
}

class Middle
{
    public Inner $inner;

    public function __construct()
    {
        $this->inner = new Inner();
    }
}

class Outer
{
    public Middle $middle;

    public function __construct()
    {
        $this->middle = new Middle();
    }
}

function dfb_source(): string { // DFB-SOURCE: nested-access-path-input
    return "tainted";
}

function dfb_sink(string $value): void {} // DFB-SINK: nested-access-path-sink

function run(): void {
    $outer = new Outer();
    $outer->middle->inner->value = dfb_source(); // DFB-WITNESS: nested-access-path-store
    $outer->middle->inner->other = "clean";
    dfb_sink($outer->middle->inner->other);
}

File: cases/taint/c/nested-access-path-negative/nested_access_path_negative.c
struct Inner {
    int value;
    int other;
};

struct Middle {
    struct Inner inner;
};

struct Outer {
    struct Middle middle;
};

int dfb_source(void) { // DFB-SOURCE: nested-access-path-input
    return 1;
}

void dfb_sink(int value) {} // DFB-SINK: nested-access-path-sink

void run(void) {
    struct Outer outer;
    outer.middle.inner.value = dfb_source(); // DFB-WITNESS: nested-access-path-store
    outer.middle.inner.other = 0;
    dfb_sink(outer.middle.inner.other);
}
